include/mxnet/c_api.h

```diff
@@ -1153,19 +1153,23 @@ MXNET_DLL int MXKVStoreBarrier(KVStoreHandle handle);
  * \brief the prototype of a server controller
  * \param head the head of the command
  * \param body the body of the command
+ * \param controller_handle helper handle for implementing controller
  */
 typedef void (MXKVStoreServerController)(int head,
-                                         const char* body);
+                                         const char *body,
+                                         void *controller_handle);
```

Comment:
is that necessary? another way may be pack the handle into the controller. for example, in c++

```
controller = [handle](int head, const char* body) { handle->xxx(); } 
```

does a similar solution work for you?